SIPS/sip-14.md

```diff
@@ -0,0 +1,109 @@
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Snap Manifest
+
+This SIP adds a new field to the Snap manifest called `dynamicPermissions`.
+This field can be used in tandem with the existing `initialPermissions`, but keep in mind that permissions in this field are not granted by installation: They MUST be requested when needed. The field follows the same format as `initialPermissions`.
+
+The new field can be specified as follows in a `snap.manifest.json` file:
+
+```json
+{
+  "initialPermissions": {
+    "endowment:transaction-insight": {}
+  },
+  "dynamicPermissions": {
+    "snap_dialog": {},
+    "snap_getBip44Entropy": [
+      {
+        "coinType": 1
+      },
+      {
+        "coinType": 3
+      }
+    ]
+  }
+}
+```
+
+### Permission caveats and merging
+
+In this initial version, duplicated permissions in `initialPermissions` and `dynamicPermissions` MUST NOT be allowed. A permission MUST only be able to exist in one of the manifest fields.
+
+Furthermore, permissions specified in `dynamicPermissions` MUST contain the caveats that will be requested at runtime and the permission request MUST fully match the caveats specified in the manifest.
+
+This MAY change in a future SIP.
+
+### RPC Methods
+
+This SIP proposes the following RPC methods to manage the dynamic permissions:
+
+#### snap_requestPermissions
+
+This RPC method SHOULD function as a subset of the existing `wallet_requestPermissions` RPC method and take the same parameters and have the same return value.
+
+This RPC method MUST prompt the user to get consent for any requested permissions and MUST validate that the requested permissions are specified in the manifest before continuing its execution (including caveats matching).
+
+#### snap_getPermissions
+
+This RPC method SHOULD be an alias for `wallet_getPermissions`, MAY be used by the snap for verifying whether it already has the permissions needed for operating. The return value and parameters SHOULD match the existing specification.
+
+#### snap_revokePermissions
+
+This RPC method SHOULD take a similar input to `wallet_requestPermissions`, an object keyed with permission names, where the values may contain caveats if applicable.
+
+For example:
+
+```json
+{
+  "method": "snap_revokePermissions",
+  "params": {
+    "snap_getBip32Entropy": {
+      "caveats": [
+        {
+          "type": "permittedDerivationPaths",
+          "value": [
+            { "path": ["m", "44'", "60'"], "curve": "secp256k1" },
+            { "path": ["m", "0'", "0'"], "curve": "ed25519" }
+          ]
+        }
+      ]
+    }
+  }
+}
+```
+
+The caveat information passed SHOULD be ignored in the initial implementation of this. Instead of processing the caveats, the implementation SHOULD revoke the entire permission key. We will revisit this at a later time to make it more granular.

```

Comment:
Does it make sense to specify the caveats if it's ignored? Or is it for future compatibility?

Reply:
The idea was that it would be future proof this way. We can make it more granular as we see fit

Reply:
I would probably just remove the caveat part. It just makes it more complicated for Snaps developers to use this method, doesn't it? In the future when we revisit this, Snaps developers can just change it to only revoke certain caveats, and we revoke everything if the object is empty or something.